corehq/form_processor/submission_post.py

```diff
@@ -322,9 +330,29 @@ def run(self):
                 elif instance.is_error:
                     submission_type = 'error'
 
+            if instance.orig_id:
+                logger.info('Finished %s processing for Form %s with original id %s',
+                    submission_type, instance.form_id, instance.orig_id)
+            else:
+                logger.info('Finished %s processing for Form %s', submission_type, instance.form_id)
+
             response = self._get_open_rosa_response(instance, **openrosa_kwargs)
             return FormProcessingResult(response, instance, cases, ledgers, submission_type)
 
+    def _log_form_details(self, form):
+        logger.info('Received Form %s with %d attachments',
+            form.form_id, len(form.attachments_list))
+
+        for index, attachment in enumerate(form.attachments_list):
+            attachment_msg = 'Form %s, Attachment %s: %s'
+            attachment_props = [form.form_id, index, attachment.name]
+
+            if attachment.is_image:
+                attachment_msg = attachment_msg + ' (%d bytes)'
+                attachment_props.append(attachment.raw_content.size)
```

Comment:
Given the implementation of [`attachment.content_length`](https://github.com/dimagi/commcare-hq/blob/dbd92bed8dafd6d91e320ffc1ed9fdbb9e18a1e5/corehq/form_processor/models.py#L103), the type of `attachment.raw_content` apparently can vary. Are you sure it always has a `size` attribute? Note that the docstring says you should not use `attachment.content_length`, however maybe we should change that if it's needed here?

Reply:
Right -- I found that the hard way when I was trying to output the size on form.xml, so now it only outputs for images. I believe all images should have raw_content, but I can probably put in some safety rails. I'm a little scared that this caused build failures, so more defensive programming might be in order